instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py

```diff
@@ -68,7 +68,7 @@ async def _async_response_hook(
 ):
     span.set_attribute(
         HTTP_RESPONSE_BODY,
-        await response[2].aread(),
+        b"".join([part async for part in response[2]]),
```

Comment:
Could someone explain why in the part for part iteration is only needed for async when it is used for both in this pr?
https://github.com/encode/httpx/pull/2407/files

Reply:
Originally I had it applied the same in both just like the httpx PR. CI failed with a Pylint issue saying that the [part for part] expression is unnecessary. Pylint is right, no need to iterate through an iterator once again, .join() can accept response[2] as-is.

On the async one, we’re dealing with async code so we need to await each coroutine part to convert it into a list to pass to .join().